### PR TITLE
Remove all non essential references to the old Zope interfaces

### PR DESCRIPTION
**Pull request by adferrand**
As a follow-up to #8971, this PR removes all references to the old Zope interfaces, except the ones used to deprecate them and prepare for their removal.

In the process, some documentation and tests about the `Display` objects are simply removed since they are not relevant anymore given that they are removed from the public API.

### CODE REVIEW

**Comment by bmw** [comment]
I'm just starting to look through this PR and I'm noticing more references that should probably be cleaned up. A nonexhaustive list is the zope dependencies in the plugins the comment "`See `~.IPlugin.inject_parser_options`" in `certbot/certbot/plugins/common.py` and
```
$ git --no-pager grep IConfig
certbot-apache/certbot_apache/_internal/constants.py:"""Name of the mod_ssl config file as saved in `IConfig.config_dir`."""
certbot-apache/certbot_apache/_internal/constants.py:"""Name of the hash of the updated or informed mod_ssl_conf as saved in `IConfig.config_dir`."""
certbot-nginx/certbot_nginx/_internal/constants.py:"""Name of the mod_ssl config file as saved in `IConfig.config_dir`."""
certbot-nginx/certbot_nginx/_internal/constants.py:"""Name of the hash of the updated or informed mod_ssl_conf as saved in `IConfig.config_dir`."""
certbot/certbot/_internal/account.py:    :ivar .IConfig config: Client configuration
certbot/certbot/_internal/client.py:    :param .IConfig config: Client configuration.
certbot/certbot/_internal/client.py:    :param .IConfig config: Client configuration.
certbot/certbot/_internal/client.py:    :ivar .IConfig config: Client configuration.
certbot/certbot/_internal/constants.py:"""Defaults for CLI flags and `.IConfig` attributes."""
certbot/certbot/_internal/constants.py:"""Archive directory, relative to `IConfig.config_dir`."""
certbot/certbot/_internal/constants.py:"""Directory mode for ``.IConfig.config_dir`` et al."""
certbot/certbot/_internal/constants.py:"""Directory (relative to `IConfig.work_dir`) where backups are kept."""
certbot/certbot/_internal/constants.py:"""See `.IConfig.csr_dir`."""
certbot/certbot/_internal/constants.py:`IConfig.work_dir`)."""
certbot/certbot/_internal/constants.py:"""Directory (relative to `IConfig.config_dir`) where keys are saved."""
certbot/certbot/_internal/constants.py:"""Live directory, relative to `IConfig.config_dir`."""
certbot/certbot/_internal/constants.py:"""Temporary checkpoint directory (relative to `IConfig.work_dir`)."""
certbot/certbot/_internal/constants.py:"""Renewal configs directory, relative to `IConfig.config_dir`."""
certbot/certbot/_internal/constants.py:"""Name of the ssl_dhparams file as saved in `IConfig.config_dir`."""
certbot/certbot/_internal/constants.py:"""Name of the hash of the updated or informed ssl_dhparams as saved in `IConfig.config_dir`."""
certbot/certbot/_internal/eff.py:from certbot.interfaces import IConfig
certbot/certbot/_internal/eff.py:def prepare_subscription(config: IConfig, acc: Account) -> None:
certbot/certbot/_internal/eff.py:    :param IConfig config: Client configuration.
certbot/certbot/_internal/eff.py:def handle_subscription(config: IConfig, acc: Account) -> None:
certbot/certbot/_internal/eff.py:    :param IConfig config: Client configuration.
certbot/certbot/_internal/log.py:    :param certbot.interface.IConfig config: Configuration object
certbot/certbot/_internal/log.py:    :param certbot.interface.IConfig config: Configuration object
certbot/certbot/_internal/main.py:    zope.component.provideUtility(config, interfaces.IConfig)
certbot/certbot/_internal/renewal.py:                zope.component.provideUtility(lineage_config, interfaces.IConfig)
certbot/certbot/crypto_util.py:    config = zope.component.getUtility(interfaces.IConfig)
certbot/certbot/crypto_util.py:    config = zope.component.getUtility(interfaces.IConfig)
certbot/certbot/interfaces.py:class IConfig(zope.interface.Interface):  # pylint: disable=inherit-non-class
certbot/certbot/reverter.py:        # First, any changes found in IConfig.temp_checkpoint_dir are removed,
```
Do you think we should clean up these references in this or another PR? I personally prefer this one, but I don't have that strong of a preference.

**Comment by adferrand** [comment]
You are right, I remove all references of type `interfaces.*`, but not when the interfaces names are used alone. I remove them in this PR.

**Comment by adferrand** [comment]
Further cleanups are added for direct references to `IConfig`, `IPlugin`, `IPluginFactory`, `IAuthenticator`, `IInstaller`, `IDisplay` and `IReporter`.

File: certbot/certbot/_internal/display/obj.py [hunk 1]
```diff
@@ -532,16 +530,11 @@ def directory_select(self, message, default=None,
         return self.input(message, default, cli_flag)
 
 
-# The two following functions use "Any" for their parameter/output types. Normally interfaces from
-# certbot.interfaces would be used, but MyPy will not understand their semantic. These interfaces
-# will be removed soon and replaced by ABC classes that will be used also here for type checking.
-# TODO: replace Any by actual ABC classes once available
```
Comment (bmw) [inline certbot/certbot/_internal/display/obj.py:538]: I'm not planning on blocking this PR on it, but I wonder if we can use a union of the two display types here?

If that doesn't work for some reason, it's probably not worth worrying about it now, but if we wanted to do something else here, a few ideas I have are:

1. Define an ABC for this interface, but keep it under `_internal`.
2. Do further refactoring on how all the plumbing under `certbot.display.util` is structured. For instance, we could replace these heavyweight classes with simple structures maintaining things like where output should be written and those structured could either be used directly by the code in `certbot.display.util` or in helper functions here.

File: certbot/certbot/_internal/main.py [hunk 2]
```diff
@@ -1483,7 +1485,7 @@ def make_displayer(config: configuration.NamespaceConfig
 
     :param config: Configuration object
 
-    :returns: Display object implementing :class:`certbot.interfaces.IDisplay`
+    :returns: Display object implementing :class:`certbot.interfaces.Display`
```
Comment (bmw) [inline certbot/certbot/_internal/main.py:1488]: There is no `certbot.interfaces.Display` object.

**Review by bmw — CHANGES_REQUESTED** [review]
I have one docstring I think we should fix and one inline suggestion that you can do whatever you want with (including just ignore it), but otherwise this LGTM!

Another thing I think we should do is remove the dependency on `zope` from all of our plugins, but I don't care if that's done in this or another PR.

File: certbot/certbot/_internal/display/obj.py (hunk 1 above)
Comment (adferrand) [reply certbot/certbot/_internal/display/obj.py:538]: It works perfectly fine with an Union. I added that.

File: certbot/certbot/_internal/main.py (hunk 2 above)
Comment (adferrand) [reply certbot/certbot/_internal/main.py]: I removed the reference.

**Comment by adferrand** [comment]
For the zope dependencies, I am planning to clean things a little more on the third party dependencies in the next PR, so it can be added at that time.

**Review by bmw — APPROVED** [review]

**Pull request merged**